pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -185,6 +199,20 @@ func (o createAuthInfoOptions) validate() error {
 	if len(methods) > 1 {
 		return fmt.Errorf("You cannot specify more than one authentication method at the same time: %v", strings.Join(methods, ", "))
 	}
+	if o.embedCertData.Value() {
+		if o.clientCertificate.Value() == "" {
+			return errors.New("You must specify a client certificate to embed")
+		}
+		if o.clientKey.Value() == "" {
```

Comment:
Changed the validation to error if --embed-certs is passed without either client-key or client-cert. Setting just one or the other is fine.